### PR TITLE
(Core) Speed up PR validation on windows

### PR DESCRIPTION
**Pull request by nguerrera**
We're running up against build timeout on Windows and it's getting in the way of team productivity.

This change does the following on Windows PR validation:

- Disable docusaurus build
- Disable formatting
- Disable linting

All of these will still be checked on Linux, where the build is currently 2X as fast, and none of them are likely to find a real Windows-specific issue in the product.

This change also disables the redundant regen-ref-docs step that is already performed during website build.

Devs can also skip the slow docusaurus build on their machines by setting environment variable TYPESPEC_SKIP_DOCUSAURUS_BUILD=true.

### CODE REVIEW

**Comment by nguerrera** [comment]
Time is down from 22 minutes to 13 minutes end-to-end.

File: eng/pipelines/pull-request-common.yml [hunk 1]
```diff
@@ -55,16 +55,15 @@ steps:
...
 
   - script: node common/scripts/install-run-rush.js lint --verbose
     displayName: Lint
+    condition: ne(variables['Agent.OS'], 'Windows_NT')
 
   - script: cd packages/samples && npm run regen-samples
     displayName: Regenerate Samples
 
-  - script: node common/scripts/install-run-rush.js regen-docs
-    displayName: Regenerate Reference Docs
```
Comment (markcowl) [inline eng/pipelines/pull-request-common.yml:66]: Don't we want to still do this on linux?

**Review by markcowl — CHANGES_REQUESTED** [review]
One question, otherwise, lgtm

File: eng/pipelines/pull-request-common.yml (hunk 1 above)
Comment (nguerrera) [reply eng/pipelines/pull-request-common.yml:66]: I think this was purely redundant. As far as I can tell this reruns the regen-ref-docs that the website build also does. @allenjzhang Am I missing something?

**Review by markcowl — APPROVED** [review]

**Pull request merged**